lib/api/objectPutPart.js

```diff
@@ -315,18 +345,17 @@ function objectPutPart(authInfo, request, streamingV4Params, log,
                 'content-md5': hexDigest,
                 'content-length': size,
             };
-            return metadata.putObjectMD(mpuBucketName, partKey, omVal, {}, log,
-                err => {
-                    if (err) {
-                        log.error('error putting object in mpu bucket', {
-                            error: err,
-                            method: 'objectPutPart::metadata.putObjectMD',
-                        });
-                        return next(err, destinationBucket);
-                    }
-                    return next(null, oldLocations, objectLocationConstraint,
-                        destinationBucket, hexDigest, prevObjectSize);
-                });
+            if (process.env.TEST_API_DATA === 'true') {
```

Comment:
A less intrusive way to achieve the same could be to instrument the memory backend itself to e.g. set a flag to ask it to delete or not store the next metadata blob to be put, called from the test code itself with a specific API exposed by the memory backend, so that a subsequent GET would fail. This way you're only instrumenting test-related code and no production code needs to be instrumented, and you don't need any extra env variable.

Reply:
I like that idea! 😄 I've made it so the unit test defines an error that the API will return. It is a small change and simple interface that gives a lot of control for testing purposes. What do you think?